microsoft-identity-broker: something went wrong [1001]

### Nixpkgs version

- Unstable (25.05)

### Describe the bug

When trying to register my device using the intune-portal, I get stuck after typing my Microsoft work account and confirming. While the pop-up turns to white, journalctl shows a lot of errors happening in the microsoft-identity-broker process. I cannot determine which of the errors might be the root cause and if it is a dependency thing or a local issue. After hitting a time-out the intune-portal pop-up shows a generic error ('something went wrong [1001]').

### Steps to reproduce

1. open intune-portal
2. click the sign-in button
3. type my Microsoft work account and confirm


### Expected behaviour

I can successfully register my device with Microsoft Intune

### Screenshots

_No response_

### Relevant log output

```console
Apr 25 22:02:56 hostname-redacted microsoft-identity-broker[715232]: I/getAccounts: [2025-04-25 20:02:56 - thread_id: 28, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] Received method call from UID [1000], with correlationId [xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx].
Apr 25 22:02:56 hostname-redacted microsoft-identity-broker[715232]: I/LinuxBrokerPlatformComponents:getDbFileRootDir: [2025-04-25 20:02:56 - thread_id: 28, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] STATE_DIRECTORY is /home/redacted/.local/state/microsoft-identity-broker
Apr 25 22:02:56 hostname-redacted microsoft-identity-broker[715232]: I/LinuxBrokerPlatformComponents:getDbFileRootDir: [2025-04-25 20:02:56 - thread_id: 28, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] STATE_DIRECTORY is /home/redacted/.local/state/microsoft-identity-broker
Apr 25 22:02:56 hostname-redacted microsoft-identity-broker[715232]: I/LinuxBrokerPlatformComponents:getDbFileRootDir: [2025-04-25 20:02:56 - thread_id: 28, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] STATE_DIRECTORY is /home/redacted/.local/state/microsoft-identity-broker
Apr 25 22:02:56 hostname-redacted microsoft-identity-broker[715232]: I/BrokerUtil:getCacheRecordListFromBrokerCache: [2025-04-25 20:02:56 - thread_id: 28, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] This client ID is not known to brokerOAuth2TokenCache.
Apr 25 22:02:56 hostname-redacted microsoft-identity-broker[715232]: I/LinuxBrokerPlatformComponents:getDbFileRootDir: [2025-04-25 20:02:56 - thread_id: 28, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] STATE_DIRECTORY is /home/redacted/.local/state/microsoft-identity-broker
Apr 25 22:02:56 hostname-redacted microsoft-identity-broker[715232]: I/BrokerUtil:getCacheRecordListFromBrokerCache: [2025-04-25 20:02:56 - thread_id: 28, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] No accounts available in client app cache, trying the FOCI cache.
Apr 25 22:02:56 hostname-redacted microsoft-identity-broker[715232]: I/LinuxBrokerPlatformComponents:getDbFileRootDir: [2025-04-25 20:02:56 - thread_id: 28, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] STATE_DIRECTORY is /home/redacted/.local/state/microsoft-identity-broker
Apr 25 22:02:56 hostname-redacted microsoft-identity-broker[715232]: W/DefaultBrokerApplicationRegistry:getMetadata: [2025-04-25 20:02:56 - thread_id: 28, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] Metadata could not be found for clientId, environment: [client-id-redacted, null]
Apr 25 22:02:56 hostname-redacted microsoft-identity-broker[715232]: I/AuthSdkOperation:isAppInBrokerApplicationRegistry: [2025-04-25 20:02:56 - thread_id: 28, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] App in broker application registry: [false]
Apr 25 22:02:56 hostname-redacted microsoft-identity-broker[715232]: I/AuthSdkOperation:addDeviceAccountIfNeeded: [2025-04-25 20:02:56 - thread_id: 28, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] App in registry is allowed to access WPJ: [false]
Apr 25 22:02:56 hostname-redacted microsoft-identity-broker[715232]: I/AuthSdkOperation:addDeviceAccountIfNeeded: [2025-04-25 20:02:56 - thread_id: 28, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] is a known FoCI App: [true]
Apr 25 22:02:56 hostname-redacted microsoft-identity-broker[715232]: I/LinuxBrokerServiceOperation:getAccounts: [2025-04-25 20:02:56 - thread_id: 28, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] Received get account result for correlation id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
Apr 25 22:02:56 hostname-redacted microsoft-identity-broker[715232]: I/BrokerDBusV1Impl:getAccounts: [2025-04-25 20:02:56 - thread_id: 28, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] Sending result back to calling application for correlation id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/acquireTokenInteractively: [2025-04-25 20:03:03 - thread_id: 30, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] Received method call from UID [1000], with correlationId [xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx].
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/LinuxBrokerPlatformComponents:getDbFileRootDir: [2025-04-25 20:03:03 - thread_id: 30, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] STATE_DIRECTORY is /home/redacted/.local/state/microsoft-identity-broker
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/LinuxBrokerPlatformComponents:getDbFileRootDir: [2025-04-25 20:03:03 - thread_id: 30, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] STATE_DIRECTORY is /home/redacted/.local/state/microsoft-identity-broker
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/LinuxBrokerPlatformComponents:getDbFileRootDir: [2025-04-25 20:03:03 - thread_id: 30, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] STATE_DIRECTORY is /home/redacted/.local/state/microsoft-identity-broker
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/AuthSdkOperation:acquireToken: [2025-04-25 20:03:03 - thread_id: 30, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] Acquire token interactive invoked, api id: 141
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/BrokerControllerFactory:getPrtV3AcquireTokenController: [2025-04-25 20:03:03 - thread_id: 30, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] Broker Account not found using identifier
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/LinuxBrokerPlatformComponents:getDbFileRootDir: [2025-04-25 20:03:03 - thread_id: 30, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] STATE_DIRECTORY is /home/redacted/.local/state/microsoft-identity-broker
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/BrokerControllerFactory:getPrtV3AcquireTokenController: [2025-04-25 20:03:03 - thread_id: 30, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] Returning BrokerSsoController.
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/LocalBroadcaster:broadcast: [2025-04-25 20:03:03 - thread_id: 87, correlation_id: UNSET - ] broadcasting to alias: cancel_authorization_request
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/OAuth2JavaFxWebViewClient: [2025-04-25 20:03:03 - thread_id: 87, correlation_id: UNSET - ] Received Authorization flow cancel request from SDK
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/AuthorizationCompletionCallback: [2025-04-25 20:03:03 - thread_id: 87, correlation_id: UNSET - ] Sending result from Authorization Activity, resultCode: SDK_CANCELLED
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/LocalBroadcaster:unregisterCallback: [2025-04-25 20:03:03 - thread_id: 87, correlation_id: UNSET - ] Removing alias: cancel_authorization_request
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/LocalBroadcaster:broadcast: [2025-04-25 20:03:03 - thread_id: 87, correlation_id: UNSET - ] broadcasting to alias: return_authorization_request_result
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/AuthorizationResultFactory:createAuthorizationResult: [2025-04-25 20:03:03 - thread_id: 87, correlation_id: UNSET - ] SDK cancelled the authorization request.
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/MicrosoftStsAuthorizationResultFactory:createAuthorizationResultWithErrorResponse: [2025-04-25 20:03:03 - thread_id: 87, correlation_id: UNSET - ] Error is returned from webview redirect
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: W/Telemetry: [2025-04-25 20:03:03 - thread_id: 34, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] No telemetry observer set.
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/LocalBroadcaster:unregisterCallback: [2025-04-25 20:03:03 - thread_id: 34, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] Removing alias: return_authorization_request_result
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/CommandDispatcher:beginInteractive: [2025-04-25 20:03:03 - thread_id: 34, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] Completed interactive request for correlation id : xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx, with the status : ERROR
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: E/AuthSdkOperation:acquireToken: [2025-04-25 20:03:03 - thread_id: 26, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] Acquire token failed.
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: java.util.concurrent.ExecutionException: com.microsoft.identity.common.java.exception.ClientException: Sdk cancelled the auth flow as the app launched a new interactive auth request.
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at com.microsoft.identity.common.java.util.ResultFuture.get(ResultFuture.java:78)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at com.microsoft.identity.broker4j.broker.ipc.AuthSdkOperation.acquireToken(AuthSdkOperation.java:214)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at com.microsoft.identity.broker.client.LinuxBrokerServiceOperation.acquireTokenInteractively(LinuxBrokerServiceOperation.java:398)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at com.microsoft.identity.broker.dbus.services.impl.BrokerDBusV1Impl.acquireTokenInteractively(BrokerDBusV1Impl.java:160)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at java.base/java.lang.reflect.Method.invoke(Method.java:566)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at org.freedesktop.dbus.connections.AbstractConnection$2.run(AbstractConnection.java:816)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at java.base/java.lang.Thread.run(Thread.java:829)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: Caused by: com.microsoft.identity.common.java.exception.ClientException: Sdk cancelled the auth flow as the app launched a new interactive auth request.
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at com.microsoft.identity.common.java.controllers.ExceptionAdapter.exceptionFromAuthorizationResult(ExceptionAdapter.java:126)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at com.microsoft.identity.broker4j.broker.prt.PrtController.acquirePrt(PrtController.java:187)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at com.microsoft.identity.client.BrokerSsoController.acquireToken(BrokerSsoController.java:209)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at com.microsoft.identity.common.java.commands.InteractiveTokenCommand.execute(InteractiveTokenCommand.java:80)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at com.microsoft.identity.common.java.commands.InteractiveTokenCommand.execute(InteractiveTokenCommand.java:47)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at com.microsoft.identity.common.java.controllers.CommandDispatcher.executeCommand(CommandDispatcher.java:542)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at com.microsoft.identity.common.java.controllers.CommandDispatcher.access$100(CommandDispatcher.java:101)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at com.microsoft.identity.common.java.controllers.CommandDispatcher$5.run(CommandDispatcher.java:776)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at io.opentelemetry.context.Context.lambda$wrap$1(Context.java:212)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         ... 3 more
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/BrokerExceptionClassifier:getExceptionCategoryInternal: [2025-04-25 20:03:03 - thread_id: 26, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] Getting an error with ErrorCode:auth_cancelled_by_sdk SubErrorCode:
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/CommandDispatcher:beginInteractive:BrokerInteractiveTokenCommandParameters: [2025-04-25 20:03:03 - thread_id: 34, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] Starting request with request context: {"thread_id":"34","correlation_id":"xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx","x-client-SKU":"MSAL.xplat.Linux"}, with PublicApiId: 141
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/InteractiveRequestAcquireTokenErrorHandler:handleError: [2025-04-25 20:03:03 - thread_id: 26, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] irrecoverable category::NOT_RECOVERABLE - return error
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: E/LinuxBrokerServiceOperation:acquireTokenInteractively: [2025-04-25 20:03:03 - thread_id: 26, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] Received error for correlation id [xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx] with Error Code: auth_cancelled_by_sdk
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: com.microsoft.identity.common.java.exception.ClientException: Sdk cancelled the auth flow as the app launched a new interactive auth request.
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at com.microsoft.identity.common.java.controllers.ExceptionAdapter.exceptionFromAuthorizationResult(ExceptionAdapter.java:126)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at com.microsoft.identity.broker4j.broker.prt.PrtController.acquirePrt(PrtController.java:187)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at com.microsoft.identity.client.BrokerSsoController.acquireToken(BrokerSsoController.java:209)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at com.microsoft.identity.common.java.commands.InteractiveTokenCommand.execute(InteractiveTokenCommand.java:80)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at com.microsoft.identity.common.java.commands.InteractiveTokenCommand.execute(InteractiveTokenCommand.java:47)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at com.microsoft.identity.common.java.controllers.CommandDispatcher.executeCommand(CommandDispatcher.java:542)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at com.microsoft.identity.common.java.controllers.CommandDispatcher.access$100(CommandDispatcher.java:101)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at com.microsoft.identity.common.java.controllers.CommandDispatcher$5.run(CommandDispatcher.java:776)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at io.opentelemetry.context.Context.lambda$wrap$1(Context.java:212)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at java.base/java.lang.Thread.run(Thread.java:829)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/BrokerDBusV1Impl:acquireTokenInteractively: [2025-04-25 20:03:03 - thread_id: 26, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] Sending result back to calling application for correlation id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/CommandDispatcher:beginInteractive:BrokerInteractiveTokenCommandParameters: [2025-04-25 20:03:03 - thread_id: 34, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] {"callerUid":1000,"shouldResolveInterrupt":false,"pKeyAuthHeaderAllowed":false,"prompt":"SELECT_ACCOUNT","authorizationAgent":"WEBVIEW","isWebViewZoomEnabled":false,"isWebViewZoomControlsEnabled":false,"handleNullTaskAffinity":false,"scopes":["redacted-scope"],"authority":{},"claimsRequestJson":"{\"access_token\":{\"deviceid\":{\"essential\":true}}}","authenticationScheme":{},"forceRefresh":true,"requiredBrokerProtocolVersion":"0.0","sdkType":"MSAL_XPLAT_LINUX","clientId":"client-id-redacted","redirectUri":"redacted-uri","powerOptCheckEnabled":false,"callerPackageName":"UNSET","correlationId":"xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"}
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/LocalBroadcaster:registerCallback: [2025-04-25 20:03:03 - thread_id: 34, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] Registering alias: return_authorization_request_result
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/InteractiveTokenCommand:execute: [2025-04-25 20:03:03 - thread_id: 34, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] Executing interactive token command...
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/BrokerSsoController:acquireToken:BrokerInteractiveTokenCommandParameters: [2025-04-25 20:03:03 - thread_id: 34, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] {"callerUid":1000,"shouldResolveInterrupt":false,"pKeyAuthHeaderAllowed":false,"prompt":"SELECT_ACCOUNT","authorizationAgent":"WEBVIEW","isWebViewZoomEnabled":false,"isWebViewZoomControlsEnabled":false,"handleNullTaskAffinity":false,"scopes":["redacted-scope"],"authority":{},"claimsRequestJson":"{\"access_token\":{\"deviceid\":{\"essential\":true}}}","authenticationScheme":{},"forceRefresh":true,"requiredBrokerProtocolVersion":"0.0","sdkType":"MSAL_XPLAT_LINUX","clientId":"client-id-redacted","redirectUri":"redacted-uri","powerOptCheckEnabled":false,"callerPackageName":"UNSET","correlationId":"xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"}
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/Authority:getKnownAuthorityResult: [2025-04-25 20:03:03 - thread_id: 34, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] Performing cloud discovery
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/Authority:performCloudDiscovery: [2025-04-25 20:03:03 - thread_id: 34, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] Performing cloud discovery...
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/Authority:performCloudDiscovery: [2025-04-25 20:03:03 - thread_id: 34, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] Acquired lock.
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/Authority:getKnownAuthorityResult: [2025-04-25 20:03:03 - thread_id: 34, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] Cloud discovery complete.
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/Authority:getKnownAuthorityResult: [2025-04-25 20:03:03 - thread_id: 34, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] Cloud is known.
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/AbstractBrokerController:convertToBrokerInteractiveTokenCommandParameters:BrokerInteractiveTokenCommandParameters: [2025-04-25 20:03:03 - thread_id: 34, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] {"callerUid":1000,"shouldResolveInterrupt":false,"pKeyAuthHeaderAllowed":false,"prompt":"SELECT_ACCOUNT","authorizationAgent":"WEBVIEW","isWebViewZoomEnabled":false,"isWebViewZoomControlsEnabled":false,"handleNullTaskAffinity":false,"scopes":["redacted-scope"],"authority":{},"claimsRequestJson":"{\"access_token\":{\"deviceid\":{\"essential\":true}}}","authenticationScheme":{},"forceRefresh":true,"requiredBrokerProtocolVersion":"0.0","sdkType":"MSAL_XPLAT_LINUX","clientId":"client-id-redacted","redirectUri":"redacted-uri","powerOptCheckEnabled":false,"callerPackageName":"UNSET","correlationId":"xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"}
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/MicrosoftStsOAuth2Strategy:createAuthorizationRequestBuilder: [2025-04-25 20:03:03 - thread_id: 34, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] Creating AuthorizationRequestBuilder
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/MicrosoftStsOAuth2Strategy:createAuthorizationRequestBuilder: [2025-04-25 20:03:03 - thread_id: 34, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] Creating AuthorizationRequestBuilder...
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: W/Device:getProductVersion: [2025-04-25 20:03:03 - thread_id: 34, correlation_id: Product version is not set. - ]
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/LinuxBrokerPlatformComponents:getDbFileRootDir: [2025-04-25 20:03:03 - thread_id: 34, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] STATE_DIRECTORY is /home/redacted/.local/state/microsoft-identity-broker
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/LocalBroadcaster:registerCallback: [2025-04-25 20:03:03 - thread_id: 34, correlation_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx - ] Registering alias: cancel_authorization_request
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/OAuth2JavaFxWebViewClient: [2025-04-25 20:03:03 - thread_id: 23, correlation_id: UNSET - ] onPageStarted: URI has no auth code ('code') query parameter.
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/IdentityHttpsURLHandler:getRefreshTokenCredential: [2025-04-25 20:03:03 - thread_id: 415, correlation_id: UNSET - ] Can't find PRT header as account not present in storage.
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: Exception in thread "JavaFX Application Thread" java.lang.UnsatisfiedLinkError: 'void com.sun.glass.ui.gtk.GtkWindow._setGravity(long, float, float)'
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at com.sun.glass.ui.gtk.GtkWindow._setGravity(Native Method)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at com.sun.glass.ui.gtk.GtkWindow._setBounds(GtkWindow.java:201)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at com.sun.glass.ui.Window.setBounds(Window.java:638)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at com.sun.javafx.tk.quantum.WindowStage.setBounds(WindowStage.java:339)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at javafx.stage.Window$TKBoundsConfigurator.apply(Window.java:1554)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at javafx.stage.Window.applyBounds(Window.java:1412)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at javafx.stage.Window.adjustSize(Window.java:316)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at javafx.stage.Window$12.invalidated(Window.java:1103)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at javafx.beans.property.BooleanPropertyBase.markInvalid(BooleanPropertyBase.java:110)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at javafx.beans.property.BooleanPropertyBase.set(BooleanPropertyBase.java:145)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at javafx.stage.Window.setShowing(Window.java:1174)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at javafx.stage.Window.show(Window.java:1189)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at javafx.stage.Stage.show(Stage.java:273)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:428)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at java.base/java.security.AccessController.doPrivileged(Native Method)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:427)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:96)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$11(GtkApplication.java:277)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]:         at java.base/java.lang.Thread.run(Thread.java:829)
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/EncryptedNameValueStorage:get: [2025-04-25 20:03:03 - thread_id: 415, correlation_id: Data associated to the given key is null or empty - ]
Apr 25 22:03:03 hostname-redacted microsoft-identity-broker[715232]: I/EncryptedNameValueStorage:get: [2025-04-25 20:03:03 - thread_id: 415, correlation_id: Data associated to the given key is null or empty - ]
Apr 25 22:03:04 hostname-redacted microsoft-identity-broker[715232]: I/EncryptedNameValueStorage:get: [2025-04-25 20:03:04 - thread_id: 23, correlation_id: Data associated to the given key is null or empty - ]
Apr 25 22:03:04 hostname-redacted microsoft-identity-broker[715232]: I/EncryptedNameValueStorage:get: [2025-04-25 20:03:04 - thread_id: 23, correlation_id: Data associated to the given key is null or empty - ]
Apr 25 22:03:04 hostname-redacted microsoft-identity-broker[715232]: I/EncryptedNameValueStorage:get: [2025-04-25 20:03:04 - thread_id: 23, correlation_id: Data associated to the given key is null or empty - ]
Apr 25 22:03:05 hostname-redacted microsoft-identity-broker[715232]: I/EncryptedNameValueStorage:get: [2025-04-25 20:03:05 - thread_id: 420, correlation_id: Data associated to the given key is null or empty - ]
Apr 25 22:03:05 hostname-redacted microsoft-identity-broker[715232]: I/EncryptedNameValueStorage:get: [2025-04-25 20:03:05 - thread_id: 422, correlation_id: Data associated to the given key is null or empty - ]
Apr 25 22:03:05 hostname-redacted microsoft-identity-broker[715232]: I/EncryptedNameValueStorage:get: [2025-04-25 20:03:05 - thread_id: 425, correlation_id: Data associated to the given key is null or empty - ]
Apr 25 22:03:06 hostname-redacted microsoft-identity-broker[715232]: I/EncryptedNameValueStorage:get: [2025-04-25 20:03:06 - thread_id: 429, correlation_id: Data associated to the given key is null or empty - ]
Apr 25 22:03:07 hostname-redacted microsoft-identity-broker[715232]: I/EncryptedNameValueStorage:get: [2025-04-25 20:03:07 - thread_id: 431, correlation_id: Data associated to the given key is nul
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.22, NixOS, 25.05 (Warbler), 25.05.20250409.f675531`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.1`
 - channels(root): `"nixos-24.11"`
 - nixpkgs: `/nix/store/zd0x43g1jcav0xy0k6q9pdhy3bvif77n-source`

### Notify maintainers

@rhysmdnz 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/mi/microsoft-identity-broker/package.nix
+++ b/pkgs/by-name/mi/microsoft-identity-broker/package.nix
@@ -55,12 +55,14 @@ stdenv.mkDerivation rec {
     makeWrapper ${openjdk11}/bin/java $out/bin/microsoft-identity-broker \
       --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath buildInputs}" \
       --add-flags "-classpath $classpath" \
+      --add-flags "-Xmx256m -Xss256k -XX:+UseParallelGC -XX:ParallelGCThreads=1" \
       --add-flags "-verbose" \
       --add-flags "com.microsoft.identity.broker.service.IdentityBrokerService"
 
     makeWrapper ${openjdk11}/bin/java $out/bin/microsoft-identity-device-broker \
       --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath buildInputs}" \
       --add-flags "-classpath $classpath" \
+      --add-flags "-Xmx256m -Xss256k -XX:+UseParallelGC -XX:ParallelGCThreads=1" \
       --add-flags "-verbose" \
       --add-flags "com.microsoft.identity.broker.service.DeviceBrokerService"
 

--- a/pkgs/by-name/mi/microsoft-identity-broker/package.nix
+++ b/pkgs/by-name/mi/microsoft-identity-broker/package.nix
@@ -6,18 +6,20 @@
   openjdk11,
   jnr-posix,
   makeWrapper,
-  openjfx17,
   zip,
   nixosTests,
   bash,
+  glib,
+  xorg,
+  alsa-lib,
 }:
 stdenv.mkDerivation rec {
   pname = "microsoft-identity-broker";
   version = "2.0.1";
 
   src = fetchurl {
-    url = "https://packages.microsoft.com/ubuntu/22.04/prod/pool/main/m/microsoft-identity-broker/microsoft-identity-broker_${version}_amd64.deb";
-    hash = "sha256-v/FxtdvRaUHYqvFSkJIZyicIdcyxQ8lPpY5rb9smnqA=";
+    url = "https://packages.microsoft.com/ubuntu/24.04/prod/pool/main/m/microsoft-identity-broker/microsoft-identity-broker_${version}_amd64.deb";
+    hash = "sha256-JheJnsu1ZxJbcpt0367FqfHVdwWWvPem2fm0i8s7MGE=";
   };
 
   nativeBuildInputs = [
@@ -27,24 +29,15 @@ stdenv.mkDerivation rec {
     zip
   ];
 
+  buildInputs = [
+    glib
+    xorg.libXtst
+    alsa-lib
+  ];
+
   buildPhase = ''
     runHook preBuild
-
     rm opt/microsoft/identity-broker/lib/jnr-posix-3.1.4.jar
-    jar -uf opt/microsoft/identity-broker/lib/javafx-graphics-15-linux.jar -C ${openjfx17}/modules_libs/javafx.graphics/ libglass.so
-    jar -uf opt/microsoft/identity-broker/lib/javafx-graphics-15-linux.jar -C ${openjfx17}/modules_libs/javafx.graphics/ libglassgtk3.so
-    jar -uf opt/microsoft/identity-broker/lib/javafx-graphics-15-linux.jar -C ${openjfx17}/modules_libs/javafx.graphics/ libprism_es2.so
-    zip -d opt/microsoft/identity-broker/lib/javafx-media-15-linux.jar libavplugin-54.so
-    zip -d opt/microsoft/identity-broker/lib/javafx-media-15-linux.jar libavplugin-56.so
-    zip -d opt/microsoft/identity-broker/lib/javafx-media-15-linux.jar libavplugin-57.so
-    zip -d opt/microsoft/identity-broker/lib/javafx-media-15-linux.jar libavplugin-ffmpeg-56.so
-    zip -d opt/microsoft/identity-broker/lib/javafx-media-15-linux.jar libavplugin-ffmpeg-57.so
-    zip -d opt/microsoft/identity-broker/lib/javafx-media-15-linux.jar libavplugin-ffmpeg-58.so
-    jar -uf opt/microsoft/identity-broker/lib/javafx-media-15-linux.jar -C ${openjfx17}/modules_libs/javafx.media/ libavplugin.so
-    jar -uf opt/microsoft/identity-broker/lib/javafx-media-15-linux.jar -C ${openjfx17}/modules_libs/javafx.media/ libfxplugins.so
-    jar -uf opt/microsoft/identity-broker/lib/javafx-media-15-linux.jar -C ${openjfx17}/modules_libs/javafx.media/ libgstreamer-lite.so
-    jar -uf opt/microsoft/identity-broker/lib/javafx-media-15-linux.jar -C ${openjfx17}/modules_libs/javafx.media/ libjfxmedia.so
-
     runHook postBuild
   '';
 
@@ -60,13 +53,16 @@ stdenv.mkDerivation rec {
     classpath="$classpath:${jnr-posix}/share/java/jnr-posix-${jnr-posix.version}.jar"
     mkdir -p $out/bin
     makeWrapper ${openjdk11}/bin/java $out/bin/microsoft-identity-broker \
-      --add-flags "-classpath $classpath com.microsoft.identity.broker.service.IdentityBrokerService" \
-      --add-flags "-verbose"
-    makeWrapper ${openjdk11}/bin/java $out/bin/microsoft-identity-device-broker \
-      --add-flags "-verbose" \
+      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath buildInputs}" \
       --add-flags "-classpath $classpath" \
-      --add-flags "com.microsoft.identity.broker.service.DeviceBrokerService" \
-      --add-flags "save"
+      --add-flags "-verbose" \
+      --add-flags "com.microsoft.identity.broker.service.IdentityBrokerService"
+
+    makeWrapper ${openjdk11}/bin/java $out/bin/microsoft-identity-device-broker \
+      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath buildInputs}" \
+      --add-flags "-classpath $classpath" \
+      --add-flags "-verbose" \
+      --add-flags "com.microsoft.identity.broker.service.DeviceBrokerService"
 
     runHook postInstall
   '';
@@ -91,7 +87,7 @@ stdenv.mkDerivation rec {
         $out/bin/microsoft-identity-device-broker \
       --replace \
         /usr/lib/jvm/java-11-openjdk-amd64 \
-        ${openjdk11}/bin/java
+        ${openjdk11}
   '';
 
   passthru = {

--- a/pkgs/by-name/mi/microsoft-identity-broker/package.nix
+++ b/pkgs/by-name/mi/microsoft-identity-broker/package.nix
@@ -57,6 +57,7 @@ stdenv.mkDerivation rec {
       --add-flags "-classpath $classpath" \
       --add-flags "-Xmx256m -Xss256k -XX:+UseParallelGC -XX:ParallelGCThreads=1" \
       --add-flags "-verbose" \
+      --add-flags "-Djava.net.useSystemProxies=true" \
       --add-flags "com.microsoft.identity.broker.service.IdentityBrokerService"
 
     makeWrapper ${openjdk11}/bin/java $out/bin/microsoft-identity-device-broker \
@@ -64,6 +65,7 @@ stdenv.mkDerivation rec {
       --add-flags "-classpath $classpath" \
       --add-flags "-Xmx256m -Xss256k -XX:+UseParallelGC -XX:ParallelGCThreads=1" \
       --add-flags "-verbose" \
+      --add-flags "-Djava.net.useSystemProxies=true" \
       --add-flags "com.microsoft.identity.broker.service.DeviceBrokerService"
 
     runHook postInstall

